lib/semantics/check-omp-structure.cc

```diff
@@ -807,8 +826,14 @@ void OmpStructureChecker::Enter(const parser::OmpAlignedClause &x) {
 void OmpStructureChecker::Enter(const parser::OmpDefaultClause &) {
   CheckAllowed(OmpClause::DEFAULT);
 }
-void OmpStructureChecker::Enter(const parser::OmpDefaultmapClause &) {
+void OmpStructureChecker::Enter(const parser::OmpDefaultmapClause &x) {
   CheckAllowed(OmpClause::DEFAULTMAP);
+  using VariableCategory = parser::OmpDefaultmapClause::VariableCategory;
+  if (!std::get<std::optional<VariableCategory>>(x.t)) {
```

Comment:
I never saw the usage of `!` with `std::get<std::optional<...>>(...)` in the F18 code base before. I think it's better to write it as:
```
const auto &vc{std::get<std::optional<VariableCategory>>(x.t)};
if (!vc.has_value()) {
...
```
Adding Tim @tskeith.

Reply:
Or `if (!std::get<std::optional<VariableCategory>>(x.t).has_value()) {`.

But the original usage is fine by me too.

Reply:
Thanks, good to know!

Reply:
I'm happy to change this if there's a style we particularly want to stick to, I was just using optional how I've used it previously in other projects without really thinking. Let me know if you want me to change it or if it's fine as-is